src/Clip.cpp

```diff
@@ -245,193 +250,199 @@ Clip::~Clip()
-				frame->AddAudio(true, channel, 0, original_frame->GetAudioSamples(channel), original_frame->GetAudioSamplesCount(), 1.0);
-
-		// Get time mapped frame number (used to increase speed, change direction, etc...)
-		// TODO: Handle variable # of samples, since this resamples audio for different speeds (only when time curve is set)
-		get_time_mapped_frame(frame, new_frame_number);
-
-		// Adjust # of samples to match requested (the interaction with time curves will make this tricky)
-		// TODO: Implement move samples to/from next frame
-
-		// Apply effects to the frame (if any)
-		apply_effects(frame);
-
-		// Determine size of image (from Timeline or Reader)
-		int width = 0;
-		int height = 0;
-		if (timeline) {
-			// Use timeline size (if available)
-			width = timeline->preview_width;
-			height = timeline->preview_height;
-		} else {
-			// Fallback to clip size
-			width = reader->info.width;
-			height = reader->info.height;
+	// Now that we have re-mapped what frame number is needed, go and get the frame pointer
+	std::shared_ptr<Frame> original_frame;
+	original_frame = GetOrCreateFrame(new_frame_number);
+
+	// Copy the image from the odd field
+	if (enabled_video) {
+		frame->AddImage(
+			std::make_shared<QImage>(*original_frame->GetImage()));
+	}
+
+	// Loop through each channel, add audio
+	if (enabled_audio && reader->info.has_audio) {
+		for (int channel = 0; channel < original_frame->GetAudioChannelsCount(); channel++) {
+			frame->AddAudio(
+				true, channel, 0,
+				original_frame->GetAudioSamples(channel),
+				original_frame->GetAudioSamplesCount(), 1.0);
 		}
+	}
 
-		// Apply keyframe / transforms
-		apply_keyframes(frame, width, height);
+	// Get time mapped frame number (used to increase speed, change direction, etc...)
+	// TODO: Handle variable # of samples, since this resamples audio for different speeds (only when time curve is set)
+	get_time_mapped_frame(frame, new_frame_number);
 
-		// Cache frame
-		cache.Add(frame);
+	// Adjust # of samples to match requested (the interaction with time curves will make this tricky)
+	// TODO: Implement move samples to/from next frame
 
-		// Return processed 'frame'
-		return frame;
+	// Apply effects to the frame (if any)
+	apply_effects(frame);
+
+	// Determine size of image (from Timeline or Reader)
+	int width = 0;
+	int height = 0;
+	if (timeline) {
+		// Use timeline size (if available)
+		width = timeline->preview_width;
+		height = timeline->preview_height;
+	} else {
+		// Fallback to clip size
+		width = reader->info.width;
+		height = reader->info.height;
 	}
-	else
-		// Throw error if reader not initialized
-		throw ReaderClosed("No Reader has been initialized for this Clip.  Call Reader(*reader) before calling this method.");
+
+	// Apply keyframe / transforms
+	apply_keyframes(frame, width, height);
+
+	// Cache frame
+	cache.Add(frame);
+
+	// Return processed 'frame'
+	return frame;
```

Comment:
```suggestion
	cache.Add(frame);
	return frame;
```

No idea why I left these, either.